util/ranger/points.go

```diff
@@ -421,15 +422,45 @@ func (r *builder) buildFromNot(expr *expression.ScalarFunction) []point {
 	case ast.IsFalsity:
 		return r.buildFromIsFalse(expr, 1)
 	case ast.In:
-		rangePoints, hasNull := r.buildFromIn(expr)
+		var unsignedInt bool
+		if x, ok := expr.GetArgs()[0].(*expression.Column); ok {
+			unsignedInt = mysql.HasUnsignedFlag(x.RetType.Flag) && mysql.IsIntegerType(x.RetType.Tp)
+		}
+		inPoints, hasNull := r.buildFromIn(expr)
 		if hasNull {
 			return nil
 		}
-		retRangePoints := make([]point, 0, len(rangePoints)+2)
+		var rangePoints []point
+		if unsignedInt {
```

Comment:
Please add a few comments for the following logic.

Reply:
ok